syntax = "proto3";

message ImageAndOwner {
  BytesImage image = 1;
  string owner = 2;
}

message BytesImage {
  repeated uint32 bytes = 1;
  int32 height = 2;
  int32 width = 3;
}

message Confidence {
  bool result = 1;
  float probability = 2;
}

message ImageOwner {
  string owner = 1;
  Confidence confidence = 2;
}

message HandOwners {
  repeated string names = 1;
}

message Empty {}

service ImageProcessingService {
  rpc AddImageToDatabase (ImageAndOwner) returns (Empty) {}
  rpc GetImageOwner (BytesImage) returns (ImageOwner) {}
  rpc IsImageOwnedByOwner (ImageAndOwner) returns (Confidence) {}
  rpc GetHandOwners (Empty) returns (HandOwners) {}
  rpc ResetHandOwners (Empty) returns (Empty) {}
}